packages/service-library/tests/redis/test_decorators.py

```diff
@@ -0,0 +1,310 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+# pylint:disable=protected-access
+
+
+import asyncio
+import datetime
+from collections.abc import Awaitable, Callable
+from typing import Final
+
+import pytest
+from faker import Faker
+from servicelib.redis import CouldNotAcquireLockError, RedisClientSDK, exclusive
+from servicelib.redis._decorators import (
+    _EXCLUSIVE_AUTO_EXTEND_TASK_NAME,
+    _EXCLUSIVE_TASK_NAME,
+)
+from servicelib.redis._errors import LockLostError
+from servicelib.utils import limited_gather, logged_gather
+
+pytest_simcore_core_services_selection = [
+    "redis",
+]
+pytest_simcore_ops_services_selection = [
+    "redis-commander",
+]
+
+
+def _assert_exclusive_tasks_are_cancelled(lock_name: str, func: Callable) -> None:
+    assert _EXCLUSIVE_AUTO_EXTEND_TASK_NAME.format(redis_lock_key=lock_name) not in [
+        t.get_name() for t in asyncio.tasks.all_tasks()
+    ], "the auto extend lock task was not properly stopped!"
+    assert _EXCLUSIVE_TASK_NAME.format(func_name=func.__name__) not in [
+        t.get_name() for t in asyncio.tasks.all_tasks()
+    ], "the exclusive task was not properly stopped!"
+
+
+async def _is_locked(redis_client_sdk: RedisClientSDK, lock_name: str) -> bool:
+    lock = redis_client_sdk.redis.lock(lock_name)
+    return await lock.locked()
+
+
+def _exclusive_sleeping_task(
+    redis_client_sdk: RedisClientSDK | Callable[..., RedisClientSDK],
```

Comment:
why you use callable alternative here. As far as i can see you only use instances of the sdk in the tests

Reply:
I use both. but it might be I moved that one away from somewhere I was using the callable. I'll check

Reply:
no I do use it in `test_exclusive_decorator_with_client_builder`